New module - s3_logging

wimnat opened the pull request:
New module to manage logging option of s3 buckets

bcoca commented:
cc @ansible/aws

bcoca commented:
cc @ansible/aws

gregdek commented:
cc @ansible/aws

rickmendes commented on cloud/amazon/s3_logging.py:
```diff
@@ -0,0 +1,185 @@
+    description:
+      - "The bucket to log to."
+    required: true
+    default: null
```
According to http://docs.ansible.com/developing_modules.html#module-checklist, default is not needed when required is true.

rickmendes commented on cloud/amazon/s3_logging.py:
```diff
@@ -0,0 +1,185 @@
+
+- name: Remove logging on an s3 bucket
+  s3_logging:
+    name: mywebsite.com
```
missing: target_bucket: mylogs here

ghost commented:
@wimnat I did a some quick testing with this one and it works as advertised. Just a couple of documentation updates needed. The module is requiring target_bucket on absent so I think it should be added to the example.

wimnat commented:
@rickmendes I've removed the target_bucket requirement and updated the doc to mention it is required for state=present.  I didn't want to have state=absent dependent on target_bucket.

ghost commented:
@wimnat That works for me. It makes more sense to me to not have it required for state-absent. I give this a +1.

bcoca commented on cloud/amazon/s3_logging.py:
```diff
@@ -0,0 +1,185 @@
+    description:
+      - "The prefix that should be prepended to the generated log files written to the target_bucket."
+    required: false
+    default: no
```
i'm guessing the default here is null or is 'no' the prefix that gets added?

wimnat replied:
Should be an empty string rather than null.

bcoca commented on cloud/amazon/s3_logging.py:
```diff
@@ -0,0 +1,185 @@
+    description:
+      - "Name of the s3 bucket."
+    required: true
+    default: null
```
no default needed when required is true

bcoca commented on cloud/amazon/s3_logging.py:
```diff
@@ -0,0 +1,185 @@
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            name = dict(required=True, default=None),
```
default is never used here as you require the input
